File: prisma/refs/java/container-with-most-water.java
class Solution {
    public int maxArea(int[] heights) {
        int l = 0, r = heights.length - 1, best = 0;
        while (l < r) {
            best = Math.max(best, (r - l) * Math.min(heights[l], heights[r]));
            if (heights[l] < heights[r]) l++;
            else r--;
        }
        return best;
    }
}

File: prisma/refs/py/container-with-most-water.py
def max_area(heights):
    l, r, best = 0, len(heights) - 1, 0
    while l < r:
        best = max(best, (r - l) * min(heights[l], heights[r]))
        if heights[l] < heights[r]:
            l += 1
        else:
            r -= 1
    return best
